### DISCUSSION TITLE
Advanced Security Push Protection for Open Source Projects

### DISCUSSION THREAD

**Discussion by ThindalTV** [community/community#29057] [Code Security]
Hi all!

While Advanced Security with automatic emails & notifications being sent when pushing secrets & keys to github is available for all public repositories, I can't find any way of activating actual Push Protection where github will block a push if it contains keys. Is this possible or is that 100% a paid feature that's not available for public repositories?

Asking for a friend. After getting a ton of notifications when I messed up.

**Comment by GuillermoFarias** [comment]
You can use pre-commit https://pre-commit.com/, tools like [Husky](https://typicode.github.io/husky/#/) help prevent that kind of thing when you create a commit, a lot before thinking about doing the push you already have the alert
